From 06959f83da6a62919cbc03ec75c5919c0531e9e7 Mon Sep 17 00:00:00 2001
From: Oskari Lemmela <oskari@lemmela.net>
Date: Sun, 5 Jul 2020 12:32:34 +0300
Subject: [PATCH] configs: pine64-lts: Enable SPI-FLASH

Pine64 LTS has winbond SPI-FLASH, so enable the same in defconfig
and add aliases for spi0 in -u-boot.dtsi

Signed-off-by: Oskari Lemmela <oskari@lemmela.net>
---
 arch/arm/dts/sun50i-a64-pine64-lts-u-boot.dtsi | 12 ++++++++++++
 configs/pine64-lts_defconfig                   |  1 +
 2 files changed, 13 insertions(+)
 create mode 100644 arch/arm/dts/sun50i-a64-pine64-lts-u-boot.dtsi

diff --git a/arch/arm/dts/sun50i-a64-pine64-lts-u-boot.dtsi b/arch/arm/dts/sun50i-a64-pine64-lts-u-boot.dtsi
new file mode 100644
index 0000000000..02b1ae046e
--- /dev/null
+++ b/arch/arm/dts/sun50i-a64-pine64-lts-u-boot.dtsi
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * Copyright (C) 2018 Jagan Teki <jagan@amarulasolutions.com>
+ */
+
+#include "sunxi-u-boot.dtsi"
+
+/ {
+	aliases {
+		spi0 = &spi0;
+	};
+};
diff --git a/configs/pine64-lts_defconfig b/configs/pine64-lts_defconfig
index ef108a1a31..7b6d5f32c0 100644
--- a/configs/pine64-lts_defconfig
+++ b/configs/pine64-lts_defconfig
@@ -8,6 +8,7 @@ CONFIG_DRAM_ZQ=3881949
 CONFIG_MMC0_CD_PIN=""
 CONFIG_MMC_SUNXI_SLOT_EXTRA=2
 CONFIG_SPL_SPI_SUNXI=y
+CONFIG_SPI_FLASH_WINBOND=y
 # CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
 CONFIG_DEFAULT_DEVICE_TREE="sun50i-a64-pine64-lts"
 CONFIG_SUN8I_EMAC=y
-- 
2.25.1

